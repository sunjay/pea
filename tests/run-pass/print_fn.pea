fn print_fn0(f: fn()) {
    println!(f);
}

fn print_fn1(f: fn(i64) -> i64) {
    println!(f);
}

fn print_fn2(f: fn(i64, ()) -> i64) {
    println!(f);
}

fn print_fn3(f: fn(i64, (), [[i64]]) -> i64) {
    println!(f);
}

fn foo() {}

fn bar(x: i64) -> i64 {
    x * 2
}

fn spam(x: i64, y: ()) -> i64 {
    x * 2
}

fn ok(x: i64, y: (), z: [[i64]]) -> i64 {
    x * 2
}

fn main() {
    print_fn0(foo);
    print_fn1(bar);
    print_fn2(spam);
    print_fn3(ok);
}
