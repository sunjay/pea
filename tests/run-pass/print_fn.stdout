<fn foo>
<fn bar>
<fn spam>
<fn ok>
